TrackBird.cpp

```diff
@@ -177,58 +163,53 @@ int TrackBird::InitializeBird(TrackSYSCONFIG *sysconfig)
+				errorCode = GetSensorParameter(i, FILTER_AC_WIDE_NOTCH, &sysconfigstatus.filter_wide, sizeof(sysconfigstatus.filter_wide));
+				errorCode = GetSensorParameter(i, FILTER_AC_NARROW_NOTCH, &sysconfigstatus.filter_narrow, sizeof(sysconfigstatus.filter_narrow));
+				errorCode = GetSensorParameter(i, FILTER_DC_ADAPTIVE, &sysconfigstatus.filter_dc, sizeof(sysconfigstatus.filter_dc));
+				errorCode = GetSensorParameter(i, FILTER_ALPHA_PARAMETERS, &sysconfig->alpha_parameters, sizeof(sysconfig->alpha_parameters));
 
 				std::cerr << "trakSTAR Original Filter Status: " << sysconfigstatus.filter_wide << " " 
 															 	 << sysconfigstatus.filter_narrow << " " 
 																 << sysconfigstatus.filter_dc << " " 
 																 << sysconfig->alpha_parameters.alphaOn << " "
 																 <<std::endl;
 
-				if (sysconfig->filter_dc <= 0.0f) // && i == 2
+				if (sysconfig->filter_dc <= 0.0f)
 					sysconfig->alpha_parameters.alphaOn = false;
 
 				//set status of all the filters
-				errorCode = SetSensorParameter(i,FILTER_AC_WIDE_NOTCH,&sysconfig->filter_wide, sizeof(sysconfig->filter_wide));
-				errorCode = SetSensorParameter(i,FILTER_AC_NARROW_NOTCH,&sysconfig->filter_narrow, sizeof(sysconfig->filter_narrow));
-				errorCode = SetSensorParameter(i,FILTER_DC_ADAPTIVE,&sysconfig->filter_dc, sizeof(sysconfig->filter_dc));
-				errorCode = SetSensorParameter(i,FILTER_ALPHA_PARAMETERS,&sysconfig->alpha_parameters,sizeof(sysconfig->alpha_parameters));
+				errorCode = SetSensorParameter(i, FILTER_AC_WIDE_NOTCH,&sysconfig->filter_wide, sizeof(sysconfig->filter_wide));
+				errorCode = SetSensorParameter(i, FILTER_AC_NARROW_NOTCH,&sysconfig->filter_narrow, sizeof(sysconfig->filter_narrow));
+				errorCode = SetSensorParameter(i, FILTER_DC_ADAPTIVE,&sysconfig->filter_dc, sizeof(sysconfig->filter_dc));
+				errorCode = SetSensorParameter(i, FILTER_ALPHA_PARAMETERS,&sysconfig->alpha_parameters, sizeof(sysconfig->alpha_parameters));
 
-				errorCode = GetSensorParameter(i,FILTER_AC_WIDE_NOTCH,&sysconfig->filter_wide, sizeof(sysconfig->filter_wide));
-				errorCode = GetSensorParameter(i,FILTER_AC_NARROW_NOTCH,&sysconfig->filter_narrow, sizeof(sysconfig->filter_narrow));
-				errorCode = GetSensorParameter(i,FILTER_DC_ADAPTIVE,&sysconfig->filter_dc, sizeof(sysconfig->filter_dc));
-				errorCode = GetSensorParameter(i,FILTER_ALPHA_PARAMETERS,&sysconfig->alpha_parameters,sizeof(sysconfig->alpha_parameters));
+				errorCode = GetSensorParameter(i, FILTER_AC_WIDE_NOTCH,&sysconfig->filter_wide, sizeof(sysconfig->filter_wide));
+				errorCode = GetSensorParameter(i, FILTER_AC_NARROW_NOTCH,&sysconfig->filter_narrow, sizeof(sysconfig->filter_narrow));
+				errorCode = GetSensorParameter(i, FILTER_DC_ADAPTIVE,&sysconfig->filter_dc, sizeof(sysconfig->filter_dc));
+				errorCode = GetSensorParameter(i, FILTER_ALPHA_PARAMETERS,&sysconfig->alpha_parameters, sizeof(sysconfig->alpha_parameters));
 
```

Comment:
They're not the same. The first is a Get call, to get the current status of all the settings. The second is the Set, which actually updates the settings. Then there is a final Get call to check that the setting changes went through properly.